accord-core/src/main/java/accord/local/CommandStores.java

```diff
@@ -416,6 +422,35 @@ public synchronized void shutdown()
             shard.store.shutdown();
     }
 
+    public CommandStore from(Routables<?, ?> routables)
+    {
+        return from(routables::intersects);
+    }
+
+    public CommandStore from(RoutingKey key)
+    {
+        return  from(ranges -> ranges.contains(key));
+    }
+
+    private CommandStore from(Predicate<Ranges> fn)
+    {
+        ShardHolder[] shards = current.shards;
+        for (ShardHolder holder : shards)
+        {
+            if (fn.test(holder.ranges().currentRanges()))
+                return holder.store;
+        }
+        return any();
+    }
+
+    @VisibleForTesting
+    public CommandStore any()
```

Comment:
I think I can get rid of this.  This used to play a bigger role but after a few refactors its now only used as the fallback case with 2 test cases using... I will look to see if I can move the test usage off `CommandStore` and into the new global executor for `BurnTest`...  that would allow this method to become private

> Even having the lookup by key/range/routable is somewhat undesirable because in so many cases you already know what the correct command store and should use that.
that's for the case where you are not running in a CommandStore, such as `accord.coordinate.Coordinate#start` which runs in C* space (executed by `TransactionStatement`)

Reply:
I tried getting rid of the 2 other test usage and both need this... `MessageTask` looks fine to get rid of (testing now), but `accord.burn.TopologyUpdates.CommandSync#process` needs to run inside a `CommandStore` and there isn't anything really to select which one... so need a random one...